test/Tests/src/Data/Numbers_Spec.enso

```diff
@@ -797,4 +799,46 @@ spec =
             3.truncate . should_equal 3
             -3.truncate . should_equal -3
 
+    Test.group "BigInts" <|
```

Comment:
Can we also test
```
x = to_java_bigint 10 # Java Big Integer
y = 10
z = 10.0
w = to_js_bigint 10

[x, y, z, w].each a->
    [x, y, z, w].each b->
		a+b . should_equal 20
```
?

I imagine for full BigInteger support we want these not to only work within a single type but also be cross-compatible with each other (like EnsoBigInteger and Small_Integer (long) and double are).

Reply:
As you mentioned that if a BigInteger is a power of 2 and not "too large" it will `fitInDouble`. Can we also do a check there where we generate such BigInteger in Java and 'transfer' it to Enso and check `: Integer` for it? I'd like to know if this works or not.

Reply:
> Can we have a `.is_a Integer` checks on these values? Will `BigInteger` satisfy the Enso `: Integer` bound?

Done in [b61860f](https://github.com/enso-org/enso/pull/7420/commits/b61860fe5681a58526b34e5388b53defe05b677f).